event_listener/dfusion_db/snapp_event_receiver.py

```diff
@@ -132,4 +132,36 @@ def save_parsed(self, order: Order) -> None:
             self.database.write_order(order)
         except AssertionError as exc:
             logging.critical(
-                "Failed to record Deposit [{}] - {}".format(exc, order))
+                "Failed to record Order [{}] - {}".format(exc, order))
+
+
+class AuctionSettlementReceiver(SnappEventListener):
+    def save(self, event: Dict[str, Any], block_info: Dict[str, Any]) -> None:
+        self.save_parsed(AuctionSettlement.from_dictionary(event))
+
+    def save_parsed(self, settlement: AuctionSettlement) -> None:
+        try:
+            self.__update_accounts(settlement)
+        except AssertionError as exc:
+            logging.critical(
+                "Failed to record Settlement [{}] - {}".format(exc, settlement))
+
+    def __update_accounts(self, settlement: AuctionSettlement) -> None:
+        balances = self.database.get_account_state(settlement.state_index - 1).balances.copy()
+        num_tokens = self.database.get_num_tokens()
+
+        # Apply the balance updates according to settlement.prices_and_volumes
+        # Should we bother to fetch the corresponding orders?
+        # We can assume the solution took balances into account?
```

Comment:
For now, yes. 
But later we need for sure to make sure that only valid solutions are pushed into the database. But then we need a whole price submission verification tool, which goes beyond balance checking. Let's talk about this in our meeting today